interfaces/builtin/x11.go

```diff
@@ -138,6 +139,13 @@ owner /run/user/[0-9]*/mutter/Xauthority r,
 network netlink raw,
 /run/udev/data/c13:[0-9]* r,
 /run/udev/data/+input:* r,
+
+# Deny access to ICE granted by abstractions/X
```

Comment:
What is the risk of regressions here?

Reply:
Unclear, it depends on how apps react. ICE is optional so I suspect most apps won't care or notice. It's a dead part of the spec and GNOME removed support for it a long while ago.

Reply:
I think the risk is very low.  Very few applications implement the protocol, and those that do should be able to run on desktops without an XSMP session manager.

In addition, it is not clear that any snaps can successfully communicate via ICE on current distros.  libICE 1.0.10 changed where it stores the authentication cookies to `$XDG_RUNTIME_DIR/ICEauthority`:

https://gitlab.freedesktop.org/xorg/lib/libice/-/commit/b484311c929a1b64966d89da92fafce7263006e1

This change was first included in Ubuntu 19.10, after which none of my desktop sessions have written a cookie to the old `~/.ICEauthority` location.  The `<abstractions/X>` AppArmor fragment does not grant access to the new location, so it is unlikely that a snap would be able to successfully connect.

Reply:
On my focal system I do get `~/.ICEauthority`, created on the 22nd of January 2020.

Reply:
On a fresh install of focal I only see `$XDG_RUNTIME_DIR/ICEauthority` - there is `~/.ICEauthority` so perhaps @zyga yours is left from a previous install (especially since it was created before focal was released).

I also agree the risk of regressions is very low, looks good to me.

Reply:
The XSMP server should be creating a new auth cookie each time you log in.  Assuming you're not still running a login session from January, it is unlikely that there are any valid cookies in `~/.ICEauthority`.  You can compare the contents with the following commands:

    iceauth -f $XDG_RUNTIME_DIR/ICEauthority list
    iceauth -f ~/.ICEauthority list

If the file in your home directory does not contain the cookies from the `$XDG_RUNTIME_DIR` file, then it can't be used to talk to the current gnome-session instance.

Reply:
@alexmurray yes, I removed the file and it did not come back on the subsequent login. Must have been a result of keeping HOME across many releases and upgrades.